ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -83,6 +83,7 @@ export function WorkflowLogsViewer({workflow, nodeId, initialPodName, container,
     const {popup} = useContext(Context);
     const [podName, setPodName] = useState(initialPodName || '');
     const [selectedContainer, setContainer] = useState(container);
+    const [selectedNodeId, setNodeId] = useState(nodeId);
```

Comment:
```suggestion
    const [nodeId, setNodeId] = useState(initialNodeId);
```

it sounds like we want to use this everywhere instead, no?

Reply:
Did the rename to `initialNodeId` along with the `nodeId` [derivation](https://github.com/argoproj/argo-workflows/pull/12591#discussion_r1563827615)